packages/examples/fortune-v2/job-launcher/server/src/common/dynamic/index.ts

```diff
@@ -0,0 +1,138 @@
+import { DynamicModule, Provider, Type } from "@nestjs/common";
+import { ModuleMetadata } from "@nestjs/common/interfaces";
+import { firstValueFrom, interval, race, Subject } from "rxjs";
+import { first, map } from "rxjs/operators";
+
+type InjectionToken = string | symbol | Type<any>;
+
+export interface IModuleConfigFactory<T> {
+  createModuleConfig(): Promise<T> | T;
+}
+
+export interface IAsyncModuleConfig<T> extends Pick<ModuleMetadata, "imports" | "exports"> {
+  useExisting?: {
+    value: IModuleConfigFactory<T>;
+    provide?: InjectionToken;
+  };
+  useClass?: Type<IModuleConfigFactory<T>>;
+  useFactory?: (...args: any[]) => Promise<T> | T;
+  inject?: any[];
+}
+
```

Comment:
/**
 * Creates a module configuration provider based on the provided options.
 * @param provide - The injection token for the module configuration.
 * @param options - The options for creating the module configuration provider.
 * @returns An array of providers for the module configuration.
 */